sdk/go/common/resource/plugin/context.go

```diff
@@ -47,6 +47,7 @@ type Context struct {
 	tracingSpan opentracing.Span // the OpenTracing span to parent requests within.
 
 	cancelFuncs []context.CancelFunc
+	lock        *sync.Mutex // Guards cancelFuncs.
```

Comment:
Let's rename this to cancelMu or cancelLock.
There's already another mutex on this struct (DebugTraceMutex), so we should attempt ti disambiguate.